src/backends/perf_pt/mod.rs

```diff
@@ -58,6 +57,55 @@ extern "C" {
     fn perf_pt_free_tracer(tr_ctx: *const c_void) -> c_int;
 }
 
+/// A Intel PT trace, obtained via Linux Perf.
+#[allow(dead_code)]
```

Comment:
Why do we allow dead code here? I'm more curious than anything!

Reply:
In release mode, we get:
```
warning: field is never used: `len`
  --> src/backends/perf_pt/mod.rs:63:5
   |
63 |     len: u64,
   |     ^^^^^^^^
   |
   = note: #[warn(dead_code)] on by default
```

Reply:
Can we put that just on that single attribute? I don't know if it works though...

Reply:
This will change once we decode the packets, of course. We could leave the annotation off so we don't forget to remove it later?

Reply:
If it works just on that one attribute, I think it's fine. Or maybe put the `cfg(debug_assertions)` hack on the field instead?

Reply:
`#[allow(dead_code)]` works on the one field! Neat.

Reply:
Wouldn't the `cfg(debug_assertions)` thing be better? It means if we end up deleting all code that uses the attribute, we'll get a compile-time error on the field.

Reply:
We would have to delete the code which *sets* len too. Read the error again :)

Reply:
OK.